### PR TITLE
Added logger scopes to validation message handling

### PR DESCRIPTION
**Pull request by agr**
... to ensure all the messages withing scope have package and validation metadata attached.

### CODE REVIEW

**Comment by joelverhagen** [comment]
Looks good, but what does this look like in AI?

**Comment by agr** [comment]
Adding scope itself does not produce AI messages, all other log entries that are within scope, will receive all the variables reported in all of the `BeginScope` calls of all enclosing scopes as well as additional `Scope` array containing scope messages of all enclosing scopes into the `CustomDimensions` property in AI.

**Comment by chenriksson** [comment]
Nice! Seems like this would be nice for the search service index reloads, maybe?

File: src/NuGet.Services.Validation.Orchestrator/ValidationSetProcessor.cs [hunk 1]
```diff
@@ -72,50 +72,53 @@ public class ValidationSetProcessor : IValidationSetProcessor
...
             {
-                _logger.LogInformation("Processing incomplete validation {ValidationType} for {PackageId} {PackageVersion}, validation set {ValidationSetId}, {ValidationId}",
-                    packageValidation.Type,
-                    package.PackageRegistration.Id,
-                    package.NormalizedVersion,
-                    validationSet.ValidationTrackingId,
-                    packageValidation.Key);
-                var validationConfiguration = GetValidationConfiguration(packageValidation.Type);
-                if (validationConfiguration == null)
+                using (_logger.BeginScope("Incomplete {ValidationType} Key {ValidationId}", packageValidation.Type, packageValidation.Key))
```
Comment (loic-sharma) [inline src/NuGet.Services.Validation.Orchestrator/ValidationSetProcessor.cs]: Could we add the `PackageId` and `PackageVersion` as well?

File: src/NuGet.Services.Validation.Orchestrator/ValidationSetProcessor.cs [hunk 2]
```diff
@@ -127,67 +130,70 @@ public class ValidationSetProcessor : IValidationSetProcessor
             bool tryMoreValidations = false;
             foreach (var packageValidation in validationSet.PackageValidations.Where(v => v.ValidationStatus == ValidationStatus.NotStarted))
             {
-                _logger.LogInformation("Processing not started validation {ValidationType} for {PackageId} {PackageVersion}, validation set {ValidationSetId}, {ValidationId}",
+                using (_logger.BeginScope("Not started {ValidationType} Key {ValidationId}", packageValidation.Type, packageValidation.Key))
```
Comment (loic-sharma) [inline src/NuGet.Services.Validation.Orchestrator/ValidationSetProcessor.cs]: Could we get `PackageId` and `PackageVersion` as well here?

File: src/NuGet.Services.Validation.Orchestrator/ValidationSetProcessor.cs (hunk 1 above)
Comment (agr) [reply src/NuGet.Services.Validation.Orchestrator/ValidationSetProcessor.cs]: Those are added in outer scope in message handler, they will propagate down

Comment (loic-sharma) [reply src/NuGet.Services.Validation.Orchestrator/ValidationSetProcessor.cs]: Ah, right, perfect!

File: src/NuGet.Services.Validation.Orchestrator/ValidationSetProcessor.cs (hunk 2 above)
Comment (agr) [reply src/NuGet.Services.Validation.Orchestrator/ValidationSetProcessor.cs]: No need, `ValidationMessageHandler` have them added already.

**Pull request merged**